### ISSUE TITLE
add save neutral loss, add to df

### ISSUE THREAD

**Issue by volvox292** [matchms/matchms#355]
# for example:
`save_mgf_df(filename, add_loss= True, header=False,index=Feature_ID, sep='\t')`

# or 

`df= spectrums.to_df(add_loss= True,index=Feature_ID)`
`save_mgf(filename, add_loss= True)`

# I do this in a more difficult (modify the save_as_mgf )way now would be nice to have it built in. 

# save_as_mgf  that saves neutral loss

from typing import List
import pyteomics.mgf as py_mgf
from ..Spectrum import Spectrum
import numpy as np

def save_as_mgf(spectrums: List[Spectrum], filename: str):
    """Save spectrum(s) as mgf file.

    :py:attr:`~matchms.Spectrum.losses` of spectrum will not be saved.

    Example:

    .. code-block:: python

        import numpy
        from matchms import Spectrum
        from matchms.exporting import save_as_mgf

        # Create dummy spectrum
        spectrum = Spectrum(mz=numpy.array([100, 200, 300], dtype="float"),
                            intensities=numpy.array([10, 10, 500], dtype="float"),
                            metadata={"charge": -1,
                                      "inchi": '"InChI=1S/C6H12"',
                                      "precursor_mz": 222.2})

        # Write spectrum to test file
        save_as_mgf(spectrum, "test.mgf")

    Parameters
    ----------
    spectrums:
        Expected input are match.Spectrum.Spectrum() objects.
    filename:
        Provide filename to save spectrum(s).
    """
    if not isinstance(spectrums, list):
        # Assume that input was single Spectrum
        spectrums = [spectrums]

    # Convert matchms.Spectrum() into dictionaries for pyteomics
    for spectrum in spectrums:
        spectrum_dict = {"m/z array": np.append(spectrum.peaks.mz,spectrum.losses[0]),
                         "intensity array": np.append(spectrum.peaks.intensities,spectrum.losses[1]),
                         "params": spectrum.metadata}
        # Append spectrum to file
        py_mgf.write([spectrum_dict], filename)

# then parse again with my own parser, after this to df

`df = pd.DataFrame({'id': df_list_id,'adduct': df_adduct, 'spec': df_list_spec,'smiles': df_smiles, 'inchikey': df_key })`

# Thanks!

**Comment by hechth** [comment]
hi @volvox292 - the neutral losses are currently only computed on the fly, so they are not readily available to be stored.

How are you using this in your work? Maybe you can give a bit more context so we can find a solution.

**Comment by volvox292** [comment]
Hi,
sorry for the late reply.
I am still using matchms 0.11.0 I think loss calculation is different there. I tried to switch to newer versions but realized  it is impossible to load big libraries with newer versions (something was changed that lead to some kind of exponential RAM consumption). These functions would be nice to feed deep learning models like the ones I am working with at the moment, but if I cannot parse big libraries anymore it is kind of useless for me anyway.

Thanks
